projects/gradio_demo/conversation.py

```diff
@@ -0,0 +1,135 @@
+import dataclasses
+from typing import List
+
+import torch
+
+
+@dataclasses.dataclass
+class Conversation:
+    system: str
+    roles: List[str]
+    messages: List[List[str]]
+    sep: str = '###'
```

Comment:
Is the `conversation` class copied from another repository? If it is, please note at the docstring or the head of the file.

Reply:
Have added the original repository at the head of ``conversation.py``.